Description: Disable Miri native-lib feature
 The native-lib feature in Miri is a work-in-progress and it pulls in a
 bunch of bundled C dependencies.
 .
 As of right now, it simply isn't worth it to include this feature,
 especially considering the rust-miri package is intended for automated
 Rust package checks.
Author: Max Gilmour <max.gilmour@canonical.com>
Forwarded: not-needed
Last-Update: 2026-02-12
---
--- a/src/tools/miri/Cargo.toml
+++ b/src/tools/miri/Cargo.toml
@@ -32,14 +32,16 @@
 [target.'cfg(unix)'.dependencies]
 libc = "0.2"
 # native-lib dependencies
-libffi = { version = "5.0.0", optional = true }
-libloading = { version = "0.9", optional = true }
-serde = { version = "1.0.219", features = ["derive"], optional = true }
-
-[target.'cfg(target_os = "linux")'.dependencies]
-nix = { version = "0.30.1", features = ["mman", "ptrace", "signal"], optional = true }
-ipc-channel = { version = "0.20.0", optional = true }
-capstone = { version = "0.13", optional = true }
+# Ubuntu: disabled; native-lib is experimental and it pulls in a bunch
+# of vendored C dependencies
+# libffi = { version = "5.0.0", optional = true }
+# libloading = { version = "0.9", optional = true }
+# serde = { version = "1.0.219", features = ["derive"], optional = true }
+#
+# [target.'cfg(target_os = "linux")'.dependencies]
+# nix = { version = "0.30.1", features = ["mman", "ptrace", "signal"], optional = true }
+# ipc-channel = { version = "0.20.0", optional = true }
+# capstone = { version = "0.13", optional = true }
 
 [target.'cfg(all(target_os = "linux", target_pointer_width = "64", target_endian = "little"))'.dependencies]
 genmc-sys = { path = "./genmc-sys/", version = "0.1.0", optional = true }
@@ -61,12 +63,12 @@
 harness = false
 
 [features]
-default = ["stack-cache", "native-lib"]
+default = ["stack-cache"]
 genmc = ["dep:genmc-sys"]
 stack-cache = []
 expensive-consistency-checks = ["stack-cache"]
 tracing = ["serde_json"]
-native-lib = ["dep:libffi", "dep:libloading", "dep:capstone", "dep:ipc-channel", "dep:nix", "dep:serde"]
+native-lib = []
 jemalloc = []
 
 [lints.rust.unexpected_cfgs]
